communitydebate: add file upload

Rineee opened the pull request:
Finally....

fuzzylogic2000 commented on euth/communitydebate/forms.py:
```diff
@@ -1,6 +1,7 @@
 from django import forms
 
 from adhocracy4.categories import forms as category_forms
+from euth.offlinephases import widgets as offlinephase_widgets
```
This is just a small comment/suggestion: the widget should be moved to contrib then, as it is used in two apps.

fuzzylogic2000 commented on euth/communitydebate/models.py:
```diff
@@ -38,5 +40,21 @@ class Topic(module_models.Item):
     def __str__(self):
         return self.name
 
+    def save(self, *args, **kwargs):
+        self.description = transforms.clean_html_field(
+            self.description)
+        super(Topic, self).save(*args, **kwargs)
+
```
yes!

fuzzylogic2000 commented on euth/communitydebate/models.py:
```diff
@@ -4,12 +4,14 @@
+from adhocracy4.models import base, query
 from adhocracy4.modules import models as module_models
 from adhocracy4.ratings import models as rating_models
+from euth.offlinephases import validators as offlinephases_validators
```
also contrib?!

fuzzylogic2000 commented on euth/communitydebate/templates/euth_communitydebate/topic_form.html:
```diff
@@ -67,6 +68,56 @@ <h3 class="sans-serif">
+    {% with upload_forms.max_num as fileupload_max %}
+    <h3>{% trans 'Downloads' %}</h3>
+    {% getAllowedFileTypes as allowed_file_types %}
+    <p>{% blocktrans %} You can upload a maximum of {{ fileupload_max }} documents. Allowed formats are {{ allowed_file_types }}. {% endblocktrans %}</p>
```
Cool!

fuzzylogic2000 commented on euth/communitydebate/templates/euth_communitydebate/topic_form.html:
```diff
@@ -67,6 +68,56 @@ <h3 class="sans-serif">
+    {% for form in upload_forms %}
+    <div class="card card-body">
+        {% for field in form.visible_fields %}
+        {% if field.name == 'title' or field.name == 'document' %}
```
Otherwise the topic would show, right? Maybe we can add the topic as hidden field and by that circumvent the other topic=none thing in the view? But I will check that out!

fuzzylogic2000 commented:
Oh, and to resolve the rebase conflicts, you can maybe undo the last merge-conflict and redo that with a rebase to master? Hope that works! And I can have a go if you want me to!

fuzzylogic2000 commented on euth/communitydebate/views.py:
```diff
@@ -29,15 +32,28 @@ def get_queryset(self):
+    template_name = 'euth_communitydebate/topic_form.html'
+    # hack to make rendering of error messages work when topic
+    # has not been saved yet
+    object = None
```
It's nearly right! If you just move it to the post method it's the same way as it's handled in django and all is fine (and still working)!
https://github.com/django/django/blob/d36413a2eb5678dda35559a48769703acd784898/django/views/generic/edit.py#L170

Rineee commented:
@fuzzylogic2000  I moved the validators and widget to contrib, moved the object=None to the post and now rebased.. should we merge now and go on in a new branch, what do you think?

fuzzylogic2000 commented:
> @fuzzylogic2000 I moved the validators and widget to contrib, moved the object=None to the post and now rebased.. should we merge now and go on in a new branch, what do you think?

Yes! Let's do that!
